README.md

```diff
@@ -1,45 +1,140 @@
 # PostCSS Customprop Validate
 
-[PostCSS] plugin to validate default values of CSS custom properties.
+[PostCSS] plugin to validate [fallback values] of CSS custom properties.
 
-[PostCSS]: https://github.com/postcss/postcss
+Validating the fallback values of CSS custom properties manually can be error prone and challening, specially if there are many.
 
-```css
-.foo {
-    /* Input example */
-}
-```
+This plugin validates the CSS and returns custom properties with incorrect fallback values. It can also modify the source CSS file with the updated values.
 
-```css
-.foo {
-  /* Output example */
-}
+## Usage
+
+```js
+const { readFileSync } = require("fs");
+const postcss = require("postcss");
+const path = require("path");
+
+const plugin = require("./index");
+
+const cssPath = path.resolve("./sample.css");
+const css = readFileSync(cssPath, "utf-8");
+const properties = {
+  /* key and expected fallback values of custom properties */
+};
+
+postcss([plugin({ properties, write: false, callback: () => {} })])
+  .process(css, { from: cssPath })
+  .then((result) => {
+    /* handle the result */
```

Comment:
@abhinavpreetu it would be great if you show what the result object will contain here. the `handle the result` comment is not very useful.

example

```js
.then((result) => {
  console.log(result);
  /**
   * {
   *   beep: "boop",
   *   foo: "bar",
   *   bit: "bot"
   * }
   */
})
```

Reply:
The result object is defined by PostCSS. So, I can log a few properties like `result.css` and `result.wrongProps`. In the previous version, I did add them. But then removed them when I remembered one of our discussions to skip `console.log()` in examples.
When should we add `console.log()`? I was checking out node.js docs, in which some had `console.log(data)` and some had `/* handle the data here */`

Reply:
Thanks.. makes it crystal clear